### PR TITLE
collaborators added in list api

### LINKED ISSUE #328: Metadata object should return collaborators
The metadata object (returned in
/v1/file/stats
/v1/file/list
/v1/file/objectpath
/v1/file/objecttree
/v1/file/referencepath
/v1/file/refs
)

Does not list the collaborators on the given file but the metadata object on /v1/file/meta does
Our objects should be consistent across endpoints and this is important metadata that should be returned 

Add collaborators to the metadata object and return it on the endpoints above

### PR DESCRIPTION
**Pull request by princeparmar**
for list API added collaborators for each child file of the given path.

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/storage_handler.go [hunk 1]
```diff
@@ -480,6 +480,17 @@ func (fsh *StorageHandler) ListEntities(ctx context.Context, r *http.Request) (*
...
+		if child.Type == reference.DIRECTORY {
+			continue
+		}
+		// getting collaborators for sub dirs
+		collaborators, err := reference.GetCollaborators(ctx, child.ID)
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			Logger.Error("Failed to get collaborators from refID", zap.Error(err), zap.Any("ref_id", dirref.ID))
+			return nil, err
+		}
+		result.Meta["collaborators"] = append(result.Meta["collaborators"].([]reference.Collaborator), collaborators...)
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/storage_handler.go:496]: should we only show them for owner like line 480?  `clientID != allocationObj.OwnerID`? try to check it with  @stewartie4

**Review by cnlangzi — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/blobbercore/handler/storage_handler.go (hunk 1 above)
Comment (princeparmar) [reply code/go/0chain.net/blobbercore/handler/storage_handler.go:496]: updated

**Comment by cnlangzi** [comment]
@princeparmar  I guess many system_tests should be updated, please check detail on system_tests

**Review by cnlangzi — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/issue-155-remove-size-check-create-dir  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**